tests/utils/test_cycle_consistency.py

```diff
@@ -0,0 +1,299 @@
+"""Unit tests to ensure correctness of cycle triplet extraction and cycle error computation.
+
+Author: John Lambert
+"""
+import time
+from typing import Dict, List, Tuple
+
+import numpy as np
+from gtsam import Rot3, Unit3
+from scipy.spatial.transform import Rotation
+
+import gtsfm.utils.cycle_consistency as cycle_utils
+from gtsfm.two_view_estimator import TwoViewEstimationReport
+
+
+def test_extract_triplets_adjacency_list_intersection1() -> None:
+    """Ensure triplets are recovered accurately.
+
+    Consider the following undirected graph with 1 cycle:
+
+    0 ---- 1
+          /|
+         / |
+        /  |
+      2 -- 3
+           |
+           |
+           4
+    """
+    i2Ri1_dict = {
+        (0, 1): Rot3(),
+        (1, 2): Rot3(),
+        (2, 3): Rot3(),
+        (1, 3): Rot3(),
+        (3, 4): Rot3(),
+    }
+
+    for extraction_fn in [cycle_utils.extract_triplets_adjacency_list_intersection, extract_triplets_brute_force]:
+
+        triplets = extraction_fn(i2Ri1_dict)
+        assert len(triplets) == 1
+        assert triplets[0] == (1, 2, 3)
+        assert isinstance(triplets, list)
+
+
+def test_extract_triplets_adjacency_list_intersection2() -> None:
+    """Ensure triplets are recovered accurately.
+
+	Consider the following undirected graph with 2 cycles:
+
+	0 ---- 1
+	      /|\
+	     / | \
+	    /  |  \
+	  2 -- 3 -- 5
+	       |
+	       |
+	       4
+	"""
+    i2Ri1_dict = {
+        (0, 1): Rot3(),
+        (1, 2): Rot3(),
+        (2, 3): Rot3(),
+        (1, 3): Rot3(),
+        (3, 4): Rot3(),
+        (1, 5): Rot3(),
+        (3, 5): Rot3(),
+    }
+
+    for extraction_fn in [cycle_utils.extract_triplets_adjacency_list_intersection, extract_triplets_brute_force]:
+
+        triplets = extraction_fn(i2Ri1_dict)
+        assert len(triplets) == 2
+        assert triplets[0] == (1, 2, 3)
+        assert triplets[1] == (1, 3, 5)
+
+        assert isinstance(triplets, list)
+
+
+def test_compute_cycle_error_known_GT() -> None:
+    """Ensure cycle error is computed correctly within a triplet, when ground truth is known.
+
+    Imagine 3 poses, all centered at the origin, at different orientations.
+
+    Ground truth poses:
+       Let i0 face along +x axis (0 degrees in yaw)
+       Let i2 have a 30 degree rotation from the +x axis.
+       Let i4 have a 90 degree rotation from the +x axis.
+
+    However, suppose one edge measurement is corrupted (from i0 -> i4) by 5 degrees.
+    """
+    i2Ri0 = Rot3(Rotation.from_euler("y", 30, degrees=True).as_matrix())
+    i4Ri2 = Rot3(Rotation.from_euler("y", 60, degrees=True).as_matrix())
+    i4Ri0 = Rot3(Rotation.from_euler("y", 95, degrees=True).as_matrix())
+
+    cycle_nodes = [0, 2, 4]
+    i2Ri1_dict = {
+        (0, 2): i2Ri0,  # edge i
+        (2, 4): i4Ri2,  # edge j
+        (0, 4): i4Ri0,  # edge k
+    }
+
+    two_view_reports_dict = {}
+    # rest of attributes will default to None
+    two_view_reports_dict[(0, 4)] = TwoViewEstimationReport(
```

Comment:
code is repeated too much. You could define a helper inside this unit test. 
def get_report(r_error, u_error): 
and use it 3 times.

Reply:
agreed, good point. let me know if it's cleaner now, I updated it.